### PR TITLE
Allow keeping lifetime while parsing owned data 

### PR DESCRIPTION
**Pull request by jethrogb**

### CODE REVIEW

File: dcap-ql/src/quote.rs [hunk 1]
```diff
@@ -147,35 +165,36 @@ impl<'a> Quote<'a> {
...
             report_body,
-            signature: quote.into(),
+            signature: quote,
         })
     }
 }
 
 impl<'a> Quote3Signature<'a> for Quote3SignatureEcdsaP256<'a> {
-    fn parse(type_: Quote3AttestationKeyType, mut data: &'a [u8]) -> Result<Self> {
+    fn parse(type_: Quote3AttestationKeyType, data: &Cow<'a, [u8]>) -> Result<Self> {
```
[the remark this answers is not in the document]
Comment (jethrogb) [reply dcap-ql/src/quote.rs:174]: Do something like this:
```
let mut certs = vec![];
while let Some(pos) = find_next_cert_end(&data) {
    certs.push(data.take_prefix(pos));
}
```

File: dcap-ql/src/quote.rs [hunk 2]
```diff
@@ -103,12 +103,28 @@ impl<'a, T: 'a> TakePrefix for &'a [T] {
...
+                },
+            }
+        } else {
+            bail!("Unexpected end of quote")
+        }
+    }
+}
+
 pub trait Quote3Signature<'a>: Sized {
-    fn parse(type_: Quote3AttestationKeyType, data: &'a [u8]) -> Result<Self>;
```
Comment (jethrogb) [inline dcap-ql/src/quote.rs:107]: Why take `&Cow<'a, [u8]>` instead of just `Cow<'a, [u8]>`?

**Comment by raoulstrackx** [comment]
r+

**Comment by jethrogb** [comment]
bors r=raoulstrackx

**Pull request merged**